tests/foreman/api/test_provisioning.py

```diff
@@ -276,3 +277,145 @@ def test_rhel_ipxe_provisioning(
+
+
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.on_premises_provisioning
+@pytest.mark.rhel_ver_match('[^6]')
+def test_rhel_pxe_provisioning_fips_enabled(
+    request,
+    module_provisioning_sat,
+    module_sca_manifest_org,
+    module_location,
+    provisioning_host,
+    pxe_loader,
+    module_provisioning_rhel_content,
+    provisioning_hostgroup,
+    module_lce_library,
+    module_default_org_view,
+):
+    """Provision a host with host param fips_enabled set to true
+
+    :id: 9e016e1d-757a-48e7-9159-131bb65dc4ef
+
+    :steps:
+        1. Configure satellite for provisioning
+        2. Provision a host with host param fips_enabled
+        3. Check that resulting host is registered to Satellite
+        4. Check host is subscribed to Satellite
+
+    :expectedresults:
+        1. Provisioning with host param fips_enabled is successful
+        2. Host installs right version of RHEL
+        3. Satellite is able to run REX job on the fips_enabled host
+        4. Host is registered to Satellite and subscription status is 'Success'
+
+    :parametrized: yes
+
+    :BZ: 2240076
+    """
+    sat = module_provisioning_sat.sat
+    host_mac_addr = provisioning_host._broker_args['provisioning_nic_mac_addr']
+    # Verify password hashing algorithm SHA256 is set in OS used for provisioning
+    assert module_provisioning_rhel_content.os.password_hash == 'SHA256'
+
+    host = sat.api.Host(
+        hostgroup=provisioning_hostgroup,
+        organization=module_sca_manifest_org,
+        location=module_location,
+        name=gen_string('alpha').lower(),
+        mac=host_mac_addr,
+        operatingsystem=module_provisioning_rhel_content.os,
+        subnet=module_provisioning_sat.subnet,
+        host_parameters_attributes=[
+            {
+                'name': 'remote_execution_connect_by_ip',
+                'value': 'true',
+                'parameter_type': 'boolean',
+            },
+            {'name': 'fips_enabled', 'value': 'true', 'parameter_type': 'boolean'},
+        ],
+        build=True,  # put the host in build mode
+    ).create(create_missing=False)
+    # Clean up the host to free IP leases on Satellite.
+    # broker should do that as a part of the teardown, putting here just to make sure.
+    request.addfinalizer(host.delete)
+    # Start the VM, do not ensure that we can connect to SSHD
+    provisioning_host.power_control(ensure=False)
+
+    # TODO: Implement Satellite log capturing logic to verify that
+    # all the events are captured in the logs.
+
+    # Host should do call back to the Satellite reporting
+    # the result of the installation. Wait until Satellite reports that the host is installed.
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
```

Comment:
Can we combine these 2 lines into single line?
`assert host.read().build_status_label == 'Installed'`

Reply:
no, host var is used for ip `host.ip`